usautobuild/config.py

```diff
@@ -1,127 +1,171 @@
 import json
 import os
+import typing
 
+from collections.abc import Mapping
 from logging import getLogger
 from pathlib import Path
-from typing import Any, Optional
+from typing import Any, Optional, TypeVar, Union
 
-from usautobuild.exceptions import MissingRequiredEnv
-
-from .exceptions import InvalidConfigFile, MissingConfigFile
+from .exceptions import InvalidConfigFile
 
 log = getLogger("usautobuild")
 
 
+_UNSET = object()
+
+
+class Variable:
+    __slots__ = (
+        "default",
+        "type_",
+        "arg",
+        "config",
+        "env",
+    )
+
+    def __init__(
+        self,
+        default: Any = _UNSET,
+        *,
+        type_: Optional[Any] = None,
+        arg: Optional[str] = None,
+        config: Optional[str] = None,
+        env: Optional[str] = None,
+    ):
+        self.default = default
+        self.type_ = type_
```

Comment:
why is the underscore to the right?

Reply:
because it otherwise shadows builtin `type` function